pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -182,6 +183,24 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 		if cosmosdb.IsErrorStatusCode(err, http.StatusPreconditionFailed) {
 			return nil, f.validateOpenShiftUniqueKey(ctx, doc)
 		}
+
+		now := time.Now().UTC()
+		_, err = f.db.Billing.Create(ctx, &api.BillingDocument{
```

Comment:
Is it feasible for this to be done on the back end instead?  The issue here is as follows:
* OpenShiftClusters.Create succeeds
* RP crashes
* Billing record not created
-> end user gets free cluster

Reply:
yes we can move this to the backend